com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -549,15 +549,15 @@ internal void InvokeBehaviourOnLostOwnership()
         {
             for (int i = 0; i < ChildNetworkBehaviours.Count; i++)
             {
-                ChildNetworkBehaviours[i].OnLostOwnership();
+                ChildNetworkBehaviours[i].InternalOnLostOwnership();
             }
         }
 
         internal void InvokeBehaviourOnGainedOwnership()
         {
             for (int i = 0; i < ChildNetworkBehaviours.Count; i++)
             {
-                ChildNetworkBehaviours[i].OnGainedOwnership();
+                ChildNetworkBehaviours[i].InternalOnGainedOwnership();
```

Comment:
Did you consider putting the call to OnGainedOwnership here vs inside InternalOnGainedOwnership?

Reply:
Actually, the internal accessor method is what should invoke the OnGainedOwnership but not until the "internal stuff" is done first.  It makes the entire ownership flow for NetworkBehaviours occur within NetworkBehaviours.